client/dive-common/vue-utilities/prompt-service/index.ts

```diff
@@ -0,0 +1,106 @@
+/* disabled this rule for Vue.prototype.FOO = */
+/* eslint-disable no-param-reassign,func-names */
+
+import { watch } from '@vue/composition-api';
+import { VueConstructor } from 'vue';
+import Vuetify from 'vuetify/lib';
+import Prompt from './Prompt.vue';
+
+interface PromptParams {
+  title: string;
+  text: string | string[];
+  positiveButton?: string;
+  negativeButton?: string;
+  confirm?: boolean;
+}
+
+class PromptService {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private component: any;
+
+  constructor(Vue: VueConstructor) {
+    const PromptComponent = Vue.extend(Prompt);
+    const component = new PromptComponent();
+    this.component = component;
+  }
+
+  set(
+    title: string,
+    text: string | string[],
+    positiveButton: string,
+    negativeButton: string,
+    confirm: boolean,
+    resolve: (value: boolean) => void,
+  ): void {
+    this.component.title = title;
+    this.component.text = text;
+    this.component.positiveButton = positiveButton;
+    this.component.negativeButton = negativeButton;
+    this.component.confirm = confirm;
+    this.component.resolve = resolve;
+    this.component.show = true;
+  }
+
+  show({
+    title,
+    text,
+    positiveButton = 'Confirm',
+    negativeButton = 'Cancel',
+    confirm = false,
+  }: PromptParams): Promise<boolean> {
+    let resolve: (value: boolean) => void = (value: boolean) => Promise.resolve(value);
+    const p = new Promise<boolean>((_resolve: (value: boolean) => void) => {
+      resolve = _resolve;
+    });
+
+    if (!this.component.show) {
+      this.set(title, text, positiveButton, negativeButton, confirm, resolve);
+    } else {
+      const unwatch = watch(this.component.show, () => {
+        unwatch();
+        this.set(title, text, positiveButton, negativeButton, confirm, resolve);
+      });
+    }
+    return p;
+  }
+
+  visible(): boolean {
+    return this.component.show;
+  }
+
+  invisible(): boolean {
+    return !this.component.show;
+  }
+
+  hide(): void {
+    this.component.show = false;
+  }
+
+  mount(element: HTMLElement) {
+    this.component.$mount(element);
+  }
+}
+
+// in vue 3 should use provide/inject with symbol
+let promptService: PromptService;
+
+export function usePrompt() {
+  // in vue 3 should use inject instead of singleton
+  return promptService;
+}
+
+export default function (vuetify: Vuetify) {
+  return function install(Vue: VueConstructor) {
+    Vue.extend({ vuetify, ...Prompt });
```

Comment:
There are issue with this line. This currently does not properly inject vuetify into `Prompt.vue` and the colors of the prompt box are incorrect. The code in the original javascript file was
```typescript
Prompt.vuetify = vuetify
```
However, this is not valid in typescript and does not compile. Needs further investigation on a fix.